pkg/apiserver/registry/networkpolicy/ipassociation/rest.go

```diff
@@ -0,0 +1,147 @@
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/registry/rest"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+	"k8s.io/klog/v2"
+
+	"antrea.io/antrea/pkg/apis/controlplane"
+	"antrea.io/antrea/pkg/apis/crd/v1alpha2"
+	ga "antrea.io/antrea/pkg/apiserver/registry/networkpolicy/groupassociation"
+	crdv1a2informers "antrea.io/antrea/pkg/client/informers/externalversions/crd/v1alpha2"
+	"antrea.io/antrea/pkg/controller/grouping"
+	"antrea.io/antrea/pkg/controller/types"
+	"antrea.io/antrea/pkg/features"
+)
+
+type REST struct {
+	podInformer  coreinformers.PodInformer
+	eeInformer   crdv1a2informers.ExternalEntityInformer
+	ipbQuerier   ipBlockGroupAssociationQuerier
+	groupQuerier ga.GroupAssociationQuerier
+}
+
+var (
+	_ rest.Storage = &REST{}
+	_ rest.Scoper  = &REST{}
+	_ rest.Getter  = &REST{}
+)
+
+// NewREST returns a REST object that will work against API services.
+func NewREST(podQuerier coreinformers.PodInformer,
+	eeQuerier crdv1a2informers.ExternalEntityInformer,
+	ipbQuerier ipBlockGroupAssociationQuerier,
+	groupQuerier ga.GroupAssociationQuerier) *REST {
+	return &REST{podQuerier, eeQuerier, ipbQuerier, groupQuerier}
+}
+
+type ipBlockGroupAssociationQuerier interface {
+	GetAssociatedIPBlockGroups(ip net.IP) ([]types.Group, error)
+}
+
+func (r *REST) New() runtime.Object {
+	return &controlplane.IPGroupAssociation{}
+}
+
+func (r *REST) Get(ctx context.Context, name string, options *metav1.GetOptions) (runtime.Object, error) {
+	parsedIP := net.ParseIP(name)
+	if parsedIP == nil {
+		return nil, errors.NewBadRequest("not a valid IP address")
+	}
+	matchedGroups := map[controlplane.GroupReference]struct{}{}
+	addToGroupList := func(groups []types.Group) {
+		for _, g := range groups {
+			item := controlplane.GroupReference{
+				Namespace: g.SourceReference.Namespace,
+				Name:      g.SourceReference.Name,
+				UID:       g.UID,
+			}
+			matchedGroups[item] = struct{}{}
+		}
+	}
+	addAssociatedGroups := func(obj metav1.Object) {
+		groups, err := r.groupQuerier.GetAssociatedGroups(obj.GetName(), obj.GetNamespace())
+		if err != nil {
+			klog.ErrorS(err, "Failed to get associated groups", "namespace", obj.GetNamespace(), "name", obj.GetName())
+			return
+		}
+		addToGroupList(groups)
+	}
+	matchedPod := r.getAssociatedPod(parsedIP.String())
+	if matchedPod != nil {
+		addAssociatedGroups(matchedPod)
+	} else {
+		// The namespace/name might correspond to an ExternalEntity rather than a Pod
+		matchedExternalEntities := r.getAssociatedExternalEntities(parsedIP.String())
+		if matchedExternalEntities != nil && len(matchedExternalEntities) > 0 {
+			for i := range matchedExternalEntities {
+				addAssociatedGroups(matchedExternalEntities[i])
+			}
+		}
+	}
+	// Check for ipBlock group association no matter the Pod/EE query result is
+	ipBlockGroups, err := r.ipbQuerier.GetAssociatedIPBlockGroups(parsedIP)
+	if err != nil {
+		klog.ErrorS(err, "Failed to get associated ipBlock groups")
```

Comment:
```suggestion
		klog.ErrorS(err, "Failed to get associated ipBlock groups ", "ip", parsedIP)
```
Otherwise the log itself won't be much helpful to understand which one triggers it, but perhaps this error handling is not even needed if the other comment is resolved.